content/code/testing.md

```diff
@@ -24,8 +31,37 @@ The best way to go about it is to follow the [official guidelines of the Go `tes
 >   func TestT_M() { ... }
 > ```
 
-When it comes to subtests, the names of individual subtests should be lowercased and concise.
-The tests usually start with a subtest called `nominal case` which verifies that the tested component behaves as expected in a baseline situation, where no failures occur and no edge cases are handled.
+You might sometimes want to have multiple test functions for a single method, in which case the tests should be prefixed as defined above, and followed with the purpose of the test:
+
+```go
+func TestMyType_MethodDoesSomethingWhenX(t *testing.T) { ... }
+func TestMyType_MethodDoesHandleXFailure(t *testing.T) { ... }
+// And so on.
+```
+
+When it comes to subtests, the names of individual subtests should be lowercased and concise. The tests usually start
+with a subtest called `nominal case` which verifies that the tested component behaves as expected in a baseline
+situation, where no failures occur and no edge cases are handled. Subsequent tests should follow the paths through which
+the function can flow from top to bottom.
+
+With the following example:
+
+```go
+{!function_example.md!}
+```
+
+There are three possible paths through which this function can be traversed:
+
+* The nominal case, where `s.validate.Struct` and `s.index.HeightForBlock` return no errors and the function returns a
+  valid response.
```

Comment:
Is this newline accidental or to keep line shorter?

Reply:
It's accidental, I think it's my IDE's formatter that tries to keep lines below 80 characters, nice catch :D 

```suggestion
* The nominal case, where `s.validate.Struct` and `s.index.HeightForBlock` return no errors and the function returns a valid response.
```